File: native/src/main/java/com/example/app/ui/components/PlusButton.kt
package com.example.app.ui.components

import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AddCircle
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.example.app.ui.theme.DarkOrange
import com.example.app.ui.theme.MediumOrange
import com.example.app.ui.theme.White

@Composable
fun PlusButton(
    modifier: Modifier = Modifier,
    backgroundColor: Color = MediumOrange,
    contentDescription: String? = "Add",
    enabled: Boolean = true,
    onClick: () -> Unit
) {
    Button(
        enabled = enabled,
        onClick = onClick,
        colors = ButtonDefaults.buttonColors(
            containerColor = backgroundColor
        ),
        shape = RoundedCornerShape(50),
        modifier = modifier.size(height = 48.dp, width = 140.dp)
    ) {
        Icon(
            imageVector = Icons.Default.AddCircle,
            contentDescription = contentDescription,
            tint = White,
            modifier = Modifier.fillMaxSize()

        )
    }
}

File: non_native/lib/view/components/plus_button.dart
import 'package:flutter/material.dart';

import '../../util/colors.dart';

class PlusButton extends StatelessWidget {
  final Color backgroundColor;
  final String contentDescription;
  final bool enabled;
  final VoidCallback onClick;
  final double height;
  final double width;

  const PlusButton({
    super.key,
    this.backgroundColor = AppColors.mediumOrange,
    this.contentDescription = "Add",
    this.enabled = true,
    required this.onClick,
    this.height = 35.0,
    this.width = 140.0,
  });

  @override
  Widget build(BuildContext context) {
    return SizedBox(
      height: height,
      width: width,
      child: ElevatedButton(
        style: ElevatedButton.styleFrom(
          backgroundColor: backgroundColor,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(50),
          ),
        ),
        onPressed: enabled ? onClick : null, //disables the button
        child: const Icon(
          Icons.add_circle_outline,
          color: AppColors.white,
          size: 24.0,
        ),
      ),
    );
  }
}
